recipes/exiv2/all/conanfile.py

```diff
@@ -0,0 +1,119 @@
+        "with_curl": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_png": True,
+        "with_xmp": True,
+        "with_curl": False,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("libiconv/1.16")
+        if self.options.with_png:
+            self.requires("libpng/1.6.37")
+        if self.options.with_xmp:
+            self.requires("expat/2.4.1")
+        if self.options.with_curl:
+            self.requires("libcurl/7.64.1")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["EXIV2_BUILD_SAMPLES"] = False
+        self._cmake.definitions["EXIV2_BUILD_EXIV2_COMMAND"] = False
+        self._cmake.definitions["EXIV2_ENABLE_PNG"] = self.options.with_png
+        self._cmake.definitions["EXIV2_ENABLE_XMP"] = self.options.with_xmp
+        self._cmake.definitions["EXIV2_ENABLE_NLS"] = False
+        self._cmake.definitions["EXIV2_ENABLE_WEBREADY"] = self.options.with_curl
+        self._cmake.definitions["EXIV2_ENABLE_CURL"] = self.options.with_curl
+        self._cmake.definitions["EXIV2_ENABLE_SSH"] = False
+        self._cmake.definitions["EXIV2_ENABLE_DYNAMIC_RUNTIME"] = (self.settings.compiler == "Visual Studio" and
+                "MD" in self.settings.compiler.runtime)
+        # set PIC manually because of object target exiv2_int
+        self._cmake.definitions["CMAKE_POSITION_INDEPENDENT_CODE"] = self.options.get_safe("fPIC", True)
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def package(self):
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "exiv2"
+        self.cpp_info.names["cmake_find_package_multi"] = "exiv2"
+        self.cpp_info.names["pkgconfig"] = "exiv2"
```

Comment:
Aren't these the default values?

Reply:
Yes, but we add them here explicit as a signal/message to the maintainers/readers/editors of the recipe that somebody went through the build system and made the effort to recreate the cmake/pkg_config interface.

I see one issue with the `exiv2` cmake_find_package generator: exiv2 does not namespace its installed targets:
https://github.com/Exiv2/exiv2/blob/82057a6a1d2fa390fcd6ad1f24ce954915854651/src/CMakeLists.txt#L254

Reply:
If it is not aliased, should we add a `_create_cmake_module_alias_targets` machinery to drop namespace?